subprojects/docs/src/docs/userguide/taskConfiguration.adoc

```diff
@@ -0,0 +1,269 @@
+
+| api:org.gradle.api.tasks.TaskContainer#findByPath(java.lang.String)[]
+| No direct equivalent.
+2+| See `getByPath(String)` above.
+
+| api:org.gradle.api.tasks.TaskCollection#withType(java.lang.Class)[]
+| _OK_
+2+| This is OK to use because it does not require tasks to be created immediately.
+
+| `withType(java.lang.Class).getByName(java.lang.String)`
+| `withType(java.lang.Class).named(java.lang.String)`
+2+| This returns a `TaskProvider` instead of a `Task`.
+
+| api:org.gradle.api.DomainObjectCollection#withType(java.lang.Class,org.gradle.api.Action)[]
+| `withType(java.lang.Class).configureEach(org.gradle.api.Action)`
+2+| This returns `void`, so it cannot be chained.
+
+| api:org.gradle.api.DomainObjectCollection#all(org.gradle.api.Action)[]
+| api:org.gradle.api.DomainObjectCollection#configureEach(org.gradle.api.Action)[]
+2+| This returns `void`, so it cannot be chained.
+
+| api:org.gradle.api.NamedDomainObjectSet#findAll(groovy.lang.Closure)[]
+| _OK_, with issues.
+2+| Avoid calling this method. `matching(Spec)` and `configureEach(Action)` are more appropriate in most cases.
+
+| api:org.gradle.api.tasks.TaskCollection#matching(groovy.lang.Closure)[]
+| _OK_
+2+| This is OK to use because it does not require tasks to be created immediately.
+
+| api:org.gradle.api.tasks.TaskCollection#getAt(java.lang.String)[]
+| api:org.gradle.api.tasks.TaskCollection#named(java.lang.String)[]
+2+| Avoid calling this directly as it's a Groovy convenience method. The alternative returns a `TaskProvider` instead of a `Task`.
+
+| `remove(org.gradle.api.Task)`
+| _OK_, with issues.
+2+| Avoid calling this. The behavior of `remove` with the new API may change in the future.
+
+| api:org.gradle.api.tasks.TaskContainer#replace(java.lang.String)[]
+| _OK_, with issues.
+2+| Avoid calling this. The behavior of `replace` with the new API may change in the future.
+
+| api:org.gradle.api.tasks.TaskContainer#replace(java.lang.String,java.lang.Class)[]
+| _OK_, with issues.
+2+| Avoid calling this. The behavior of `replace` with the new API may change in the future.
+
+|===
+
+[[sec:new_task_gradle_profiler_scenario]]
+=== Profiling with a Gradle Profiler Scenario File
+
+The https://github.com/gradle/gradle-profiler#gradle-profiler[Gradle Profiler] is a tool to measure build times for Gradle builds in a predictable and reproducible manner. The tool automates collecting profiling and benchmark information from a Gradle build and mitigates environmental impacts to measuring build time (like JIT warmups and cached dependencies). Clone and build `gradle-profiler` locally.
+
+To measure the impact of the new API on your build, we've included a sample scenario file you can use. This scenario runs `gradle help` on your build with a special flag to enable/disable the new API to make it easier to measure improvements.
```

Comment:
We might describe why running `gradle help` is a reasonable way to measure the performance.  It might seem obvious to us, but it may not be to others.